From c7aca9b28fbf0070b86e91c1b84be48a3351be79 Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Wed, 23 Sep 2015 18:56:10 +0530
Subject: [PATCH 07/28] FOR_UPSTREAM [VPG]: drm/i915: EQ retry must be 5 times
 only

Spec requires us to retry same voltage only upto 5 times. If the sink is
not trained by this time we should fail the request and return. This patch
enforces this requirement.

Change-Id: I84455e443b799f8c22fbc56a345e8716b14c91d5
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-9749
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Reviewed-on: https://android.intel.com:443/439680
---
 drivers/gpu/drm/i915/intel_dp.c | 23 ++++++-----------------
 1 file changed, 6 insertions(+), 17 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 6d287d9..5b5fda5 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -3958,7 +3958,7 @@ void
 intel_dp_complete_link_train(struct intel_dp *intel_dp)
 {
 	bool channel_eq = false;
-	int tries, cr_tries;
+	int tries;
 	uint32_t DP = intel_dp->DP;
 	uint32_t training_pattern = DP_TRAINING_PATTERN_2;
 
@@ -3975,13 +3975,14 @@ intel_dp_complete_link_train(struct intel_dp *intel_dp)
 	}
 
 	tries = 0;
-	cr_tries = 0;
 	channel_eq = false;
 	for (;;) {
 		uint8_t link_status[DP_LINK_STATUS_SIZE];
 
-		if (cr_tries > 5) {
-			DRM_ERROR("failed to train DP, aborting\n");
+		/* Try 5 times, then exit if that fails */
+		if (tries > 4) {
+			intel_dp_link_down(intel_dp);
+			DRM_ERROR("failed equalization after 5 tries\n");
 			break;
 		}
 
@@ -3997,7 +3998,7 @@ intel_dp_complete_link_train(struct intel_dp *intel_dp)
 			intel_dp_set_link_train(intel_dp, &DP,
 						training_pattern |
 						DP_LINK_SCRAMBLING_DISABLE);
-			cr_tries++;
+			++tries;
 			continue;
 		}
 
@@ -4006,18 +4007,6 @@ intel_dp_complete_link_train(struct intel_dp *intel_dp)
 			break;
 		}
 
-		/* Try 5 times, then try clock recovery if that fails */
-		if (tries > 5) {
-			intel_dp_link_down(intel_dp);
-			intel_dp_start_link_train(intel_dp);
-			intel_dp_set_link_train(intel_dp, &DP,
-						training_pattern |
-						DP_LINK_SCRAMBLING_DISABLE);
-			tries = 0;
-			cr_tries++;
-			continue;
-		}
-
 		/* Update training set as requested by target */
 		if (!intel_dp_update_link_train(intel_dp, &DP, link_status)) {
 			DRM_ERROR("failed to update link training\n");
-- 
1.9.1

